gaussian_example/transform_base-1.0.yaml

```diff
@@ -0,0 +1,135 @@
+      Object that encapsulates properties of a transform parameter.
+    properties:
+      value:
+        description: >
+          Parameter value.
+        $ref: '#/definitions/parameter_value'
+      unit:
+        description: >
+          Parameter unit.
+        default: >
+          Parameter has no unit.
+        tag: asdf://asdf-format.org/extensions/core/tags/unit-*
+      bounds:
+        description: >
+          Inclusive upper and lower bounds of the parameter value.
+          Used during fitting.
+        default: >
+          Parameter value is unbounded.
+        $ref: '#/definitions/bounds'
+      fixed:
+        description: >
+          If true, this parameter's value should be fixed when fitting.
+        default: >
+          Parameter is not fixed.
+        type: boolean
+    required: [value]
+    additionalProperties: false
+
+  bounds:
+    description: >
+      Inclusive upper and lower bounds for a parameter or
+      input value
+    type: array
+    items:
+      type: number
+    minItems: 2
+    maxItems: 2
+
+  input:
+    description: >
+      Object that encapsulates properties of a transform input.
+    properties:
+      name:
+        description: >
+          Input name.
+        type: string
+      unit:
+        description: >
+          Input unit.
+        default: >
+          Input has no unit.
+        tag: asdf://asdf-format.org/extensions/core/tags/unit-*
+      bounds:
+        description: >
+          Inclusive upper and lower bounds of the input value.
+        default: >
+          Input value is unbounded.
+        $ref: '#/definitions/bounds'
+    required: [name]
+    additionalProperties: false
+
+  output:
+    description: >
+      Object that encapsulates properties of a transform output.
+    properties:
+      name:
+        description: >
+          Output name.
+      unit:
+        description: >
+          Output unit.
+        default: >
+          Output has no unit.
+        tag: asdf://asdf-format.org/extensions/core/tags/unit-*
+
+properties:
+  parameters:
+    description: >
+      Map of parameter names to parameter values or parameter objects.
+    additionalProperties:
+      anyOf:
+        - $ref: '#/definitions/parameter_value'
+        - $ref: '#/definitions/parameter'
+
+  inputs:
+    description: >
+      Array of input names or input objects.
+    type: array
+    items:
+      anyOf:
+        - type: string
+        - $ref: '#/definitions/input'
+
+  outputs:
+    description: >
+      Array of output names or output objects.
+    type: array
+    items:
+      anyOf:
+        - type: string
+        - $ref: '#/definitions/output'
+
+  name:
+    description: >
+      Optional user-friendly name for the transform.
+    default: >
+      Transform has no name.
+    type: string
+
+  inverse:
+    description: >
+      Explicit inverse of this transform.
+    tag: asdf://asdf-format.org/extensions/transform/tags/**
+
+required: [parameters, inputs, outputs]
```

Comment:
IMHO if `inputs` and `outputs` are not specified then we don't care about them, they can be implementation specific. OTH having them as required will force changes in existing libraries, e.g. implementation of named inputs and outputs.

Reply:
That makes sense to me.  I had assumed that absence of these properties implied "use the astropy default names".  If it really means "undefined, feel free to select whatever names you wish" then I agree that we don't need to make them required.